### PR TITLE
update request operation name

### PR DESCRIPTION
**Pull request by vishweshbankwar**
From common schema mapping
1) `httpMethod` + " " + `httpRoute`. 
2) If `httpRoute` is not available then extract the path from `httpUrl`. If path is empty then use `/`. e.g. `GET /`. 
3) If none of the attributes are available then assign Span.Name

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartA.cs [hunk 1]
```diff
@@ -70,7 +70,18 @@ internal static string GetOperationName(Activity activity, ref AzMonList partBTa
             var httpMethod = AzMonList.GetTagValue(ref partBTags, SemanticConventions.AttributeHttpMethod)?.ToString();
             if (!string.IsNullOrWhiteSpace(httpMethod))
             {
-                return $"{httpMethod} {activity.DisplayName}";
+                var httpRoute = AzMonList.GetTagValue(ref partBTags, SemanticConventions.AttributeHttpRoute)?.ToString();
+                // ASPNET instrumentation assigns route as {controller}/{action}/{id} which would result in same name for different operations.
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartA.cs:74]: nit:

```suggestion
                // ASP.NET instrumentation assigns route as {controller}/{action}/{id} which would result in the same name for different operations.
```

**Review by reyang — APPROVED** [review]
LGTM

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartA.cs [hunk 2]
```diff
@@ -70,7 +70,18 @@ internal static string GetOperationName(Activity activity, ref AzMonList partBTa
             var httpMethod = AzMonList.GetTagValue(ref partBTags, SemanticConventions.AttributeHttpMethod)?.ToString();
             if (!string.IsNullOrWhiteSpace(httpMethod))
             {
-                return $"{httpMethod} {activity.DisplayName}";
+                var httpRoute = AzMonList.GetTagValue(ref partBTags, SemanticConventions.AttributeHttpRoute)?.ToString();
+                // ASP.NET instrumentation assigns route as {controller}/{action}/{id} which would result in the same name for different operations.
+                // To work around that we will use path from httpUrl.
+                if (!string.IsNullOrEmpty(httpRoute) && !httpRoute.StartsWith("{controller}", StringComparison.OrdinalIgnoreCase))
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartA.cs:76]: Most enterprise application uses `Area` in MVC, where the routing pattern will look like `{area:exists}/{controller=Home}/{action=Index}/{id?}`
Ref: https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/areas?view=aspnetcore-5.0#areas-for-controllers-with-views

Are we worried about this scenario?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartA.cs:76]: This question should not block the PR. We could discuss this separately.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Comment by vishweshbankwar** [comment]
Did something wrong and merged another branch into this. closing this PR - will create a new one.

**Pull request closed without merging**